Remove Auto Scaling support (#42)
azavea/terraform-aws-ecs-cluster: README.md

A Terraform module to create an Amazon Web Services (AWS) EC2 Container Service (ECS) cluster.

> * [Usage](#usage)

Reviewer: I would consider adding a Table of Contents header to this section.

```suggestion
## Table of Contents

* [Usage](#usage)
```

    * [Auto Scaling](#auto-scaling)
* [Variables](#variables)
* [Outputs](#outputs)

## Usage

> This module creates a security group that gets associated with the launch configuration for the ECS cluster Auto Scaling group. By default, the security group contains no rules. In order for network traffic to flow egress or ingress (including communication with ECS itself), you must associate all of the appropriate `aws_security_group_rule` resources with the `container_instance_security_group_id` module output.

Reviewer: ```suggestion
This module creates a security group that gets associated with the launch template for the ECS cluster Auto Scaling group. By default, the security group contains no rules. In order for network traffic to flow egress or ingress (including communication with ECS itself), you must associate all of the appropriate `aws_security_group_rule` resources with the `container_instance_security_group_id` module output.
```

}

module "container_service_cluster" {

>   source = "github.com/azavea/terraform-aws-ecs-cluster?ref=3.0.0"

Author: I am anticipating this would be part of a `3.0.0` release. 

In retrospect, this might be better to include within the `3.0.0` commit of the `release/3.0.0` branch.

}
```

### Auto Scaling

This module creates an Auto Scaling group for the ECS cluster. By default, there are no Auto Scaling policies associated with this group. In order for Auto Scaling to function, you must define `aws_autoscaling_policy` resources and associate them with the `container_instance_autoscaling_group_name` module output.

> See this [article](https://segment.com/blog/when-aws-autoscale-doesn-t/) for more information on Auto Scaling, and below for example policies. 

Reviewer: ```suggestion
See this [article](https://segment.com/blog/when-aws-autoscale-doesn-t/) for more information on Auto Scaling, and below for example policies.
```

```hcl
resource "aws_autoscaling_policy" "container_instance_cpu_reservation" {

>   name                   = "asgScalingPolicyCPUReservationStagingCluster"

Reviewer: ```suggestion
  name                   = "asgScalingPolicyCPUReservation"
```

  autoscaling_group_name = "${module.container_service_cluster.container_instance_autoscaling_group_name}"
  adjustment_type        = "ChangeInCapacity"
  policy_type            = "TargetTrackingScaling"

  target_tracking_configuration {
    customized_metric_specification {
      metric_dimension {
        name  = "ClusterName"
        value = "${module.container_service_cluster.name}"
      }

      metric_name = "CPUReservation"
      namespace   = "AWS/ECS"
      statistic   = "Average"
    }

    target_value = "50.0"
  }
}

resource "aws_autoscaling_policy" "container_instance_memory_reservation" {

>   name                   = "asgScalingPolicyMemoryReservationStagingCluster"

Reviewer: ```suggestion
  name                   = "asgScalingPolicyMemoryReservation"
```

  adjustment_type        = "ChangeInCapacity"
  policy_type            = "TargetTrackingScaling"

  target_tracking_configuration {
    customized_metric_specification {
      metric_dimension {
        name  = "ClusterName"
        value = "${module.container_service_cluster.name}"
      }

      metric_name = "CPUReservation"
      namespace   = "AWS/ECS"
      statistic   = "Average"
    }

    target_value = "50.0"
  }
}

resource "aws_autoscaling_policy" "container_instance_memory_reservation" {
  name                   = "asgScalingPolicyMemoryReservation"
  autoscaling_group_name = "${module.container_service_cluster.container_instance_autoscaling_group_name}"
  adjustment_type        = "ChangeInCapacity"
  policy_type            = "TargetTrackingScaling"

  target_tracking_configuration {
    customized_metric_specification {
      metric_dimension {
        name  = "ClusterName"
        value = "${module.container_service_cluster.name}"
      }

      metric_name = "MemoryReservation"
      namespace   = "AWS/ECS"
      statistic   = "Average"
    }

    target_value = "90.0"
  }
}
```

> It's worth nothing that the [`aws_autoscaling_policy`](https://www.terraform.io/docs/providers/aws/r/autoscaling_policy.html) documentation suggests we remove `desired_capacity` from the `aws_autoscaling_group` resource when using Auto Scaling. That makes sense, because when it is present, any Terraform plan/apply cycle will reset it.

Reviewer: ```suggestion
It's worth noting that the [`aws_autoscaling_policy`](https://www.terraform.io/docs/providers/aws/r/autoscaling_policy.html) documentation suggests `desired_capacity` be removed from the `aws_autoscaling_group` resource when using Auto Scaling. Since `desired_capacity` exists in this module, users will have to dynamically override the value passed on each Terraform plan/apply cycle.
```

Author: Good catch.

> Unfortunately, removing it from the `aws_autoscaling_group` resource means removing it from the module too.

Reviewer: I would consider the suggested change above, along with the removal of the details from here down to (but not including):

> We will reevaluate...
